build/yaml/templates/versioning-steps.yml

```diff
@@ -0,0 +1,15 @@
+steps:
+  - powershell: |
+      "Deployment Ring = $(DeploymentRing)";
+      $deploymentRing = "$(DeploymentRing)";
+      $dateStamp = (Get-Date -format "yyyyMMdd");
+      $commitHash = "$(Build.SourceVersion)".SubString(0,7);
+      $vs = $dateStamp + "." + $commitHash;
+
+      if ($deploymentRing -ne "stable") {
```

Comment:
Probably worth forcing a .ToLowerInvariant() on $deploymentRing to ensure consistent casing.